src/Gemstone.IO/Collections/FileBackedLookupTable.cs

```diff
@@ -1707,17 +1710,10 @@ static FileBackedLookupTable()
         Func<Stream, TKey>? readKeyFunc = GetReadMethod<TKey>();
         Func<Stream, TValue>? readValueFunc = GetReadMethod<TValue>();
 
-        if (writeKeyAction is null || readKeyFunc is null)
-        {
-            writeKeyAction = (_, _) => throw new InvalidOperationException($"Type of TKey ({typeof(TKey).FullName}) is not write serializable.");
-            readKeyFunc = _ => throw new InvalidOperationException($"Type of TKey ({typeof(TKey).FullName}) is not read serializable.");
-        }
-
-        if (writeValueAction is null || readValueFunc is null)
-        {
-            writeValueAction = (_, _) => throw new InvalidOperationException($"Type of TValue ({typeof(TValue).FullName}) is not write serializable.");
-            readValueFunc = _ => throw new InvalidOperationException($"Type of TValue ({typeof(TValue).FullName}) is not read serializable.");
-        }
+        writeKeyAction ??= (_, _) => throw new InvalidOperationException($"Type of TKey ({typeof(TKey).FullName}) is not write serializable.");
+        readKeyFunc ??= _ => throw new InvalidOperationException($"Type of TKey ({typeof(TKey).FullName}) is not read serializable.");
+        writeValueAction ??= (_, _) => throw new InvalidOperationException($"Type of TValue ({typeof(TValue).FullName}) is not write serializable.");
+        readValueFunc ??= _ => throw new InvalidOperationException($"Type of TValue ({typeof(TValue).FullName}) is not read serializable.");
```

Comment:
Hm, I see now why you changed the messages to specify read-serializable vs write-serializable. It's a bit odd to me to support read-only or write-only lookup tables, but given that files can be used for IPC, I suppose there could be use-cases.

Reply:
It was especially relevant when debugging manual "WriteTo"/ "ReadFrom" implementations